### PR TITLE
fix: type conversions causing overflows because of infinite recursions and save transaction changes only in case of success

### PR DESCRIPTION
**Pull request by carneiro-cw**

### CODE REVIEW

File: src/eth/primitives/nonce.rs [hunk 1]
```diff
@@ -40,7 +40,7 @@ gen_newtype_from!(self = Nonce, other = u8, u16, u32, u64, u128, U256, usize, i3
 
 impl From<BigDecimal> for Nonce {
     fn from(value: BigDecimal) -> Self {
-        value.into()
+        Nonce(U256::from_dec_str(&value.to_string()).unwrap_or(U256::zero()))
```
Comment (renancloudwalk) [inline src/eth/primitives/nonce.rs:43]: Maybe I would lof this kind of failure, down the road it can make our life easier

**Review by renancloudwalk — APPROVED** [review]
Overall I don't like plain unwrap_or without logs.

File: src/eth/primitives/nonce.rs (hunk 1 above)
Comment (carneiro-cw) [reply src/eth/primitives/nonce.rs:43]: Sure, I can log the unwrap_ors

**Review by renancloudwalk — DISMISSED** [review]

**Review by dinhani-cw — DISMISSED** [review]

**Comment by arthur-cw** [comment]
@carneiro-cw just add some coments to the unwrap_or explaining why it's OK, and it's good

**Review by arthur-cw — APPROVED** [review]

**Pull request merged**